### PR TITLE
Update IDE integration to reflect Java 14 requirement

### PR DESCRIPTION
**Pull request by mark-vieira**
Instead of hardcoding the IntelliJ JDK version just use whatever `minimumCompilerVersion` is. Also, updated our contributing docs to reflect we've bumped to Java 14.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -198,7 +198,7 @@ Please follow these formatting guidelines:
 #### Editor / IDE Support
 
 IntelliJ IDEs can
-[import](https://blog.jetbrains.com/idea/2014/01/intellij-idea-13-importing-code-formatter-settings-from-eclipse/)
+[import](https://blog.jetbrains.com/idea/2014/01/intellij-idea-`13`-importing-code-formatter-settings-from-eclipse/)
```
Comment (rjernst) [inline CONTRIBUTING.md:201]: are these backticks correct?

**Review by rjernst — APPROVED** [review]
LGTM

File: CONTRIBUTING.md (hunk 1 above)
Comment (mark-vieira) [reply CONTRIBUTING.md:201]: Good catch. I have no idea where those came from. I assume IntelliJ was trying to "help" me.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Review by pugnascotia — APPROVED** [review]

**Pull request merged**